src/lib/util/metadata.ts

```diff
@@ -28,17 +28,105 @@ export interface UserscriptDefaultMetadata {
+    pathRegex = (typeof pathRegex === 'string') ? pathRegex : pathRegex.source;
 
-export function transformMBMatchURL(requireString: string): string {
-    return `*://*.musicbrainz.org/${requireString}`;
+    return `/^https?://${domainRegex}/${pathRegex}${constructSuffix(options)}/`;
 }
 
+function removeTemplateAndCreateRule(options: CreateURLRuleRegexOptions, pattern: string): string;
+function removeTemplateAndCreateRule(options: CreateURLRuleRegexOptions, template: TemplateStringsArray, ...values: string[]): string;
+function removeTemplateAndCreateRule(options: CreateURLRuleRegexOptions, pattern: string | TemplateStringsArray, ...values: string[]): string {
+    if (typeof pattern === 'string') {
+        return createURLRuleRegex(MB_DOMAIN_REGEX, pattern, options);
+    } else {
+        const fullPattern = pattern.raw[0] + pattern.raw.slice(1)
+            .map((lit, idx) => `${values[idx]}${lit}`)
+            .join('');
+        return createURLRuleRegex(MB_DOMAIN_REGEX, fullPattern, options);
+    }
+}
+
+// To be used with a tagged template literal
+export function createMBRegex(options: CreateURLRuleRegexOptions): (pattern: string | TemplateStringsArray) => string;
+export function createMBRegex(pattern: string): string;
+export function createMBRegex(pattern: TemplateStringsArray, ...values: string[]): string;
+export function createMBRegex(patternOrOptions: CreateURLRuleRegexOptions | string | TemplateStringsArray, ...values: string[]): ((pattern: string) => string) | ((pattern: TemplateStringsArray, ...values2: string[]) => string) | string {
+    // The two first branches could in practice be merged, but TypeScript struggles with that.
+    if (typeof patternOrOptions === 'string') {
+        return removeTemplateAndCreateRule({}, patternOrOptions);
+    } else if (Array.isArray(patternOrOptions)) {
+        return removeTemplateAndCreateRule({}, patternOrOptions as TemplateStringsArray, ...values);
+    } else {
+        return removeTemplateAndCreateRule.bind(undefined, patternOrOptions as CreateURLRuleRegexOptions);
+    }
+}
+
+const mb = createMBRegex;
+export const MBID_REGEX_PART = String.raw`[a-f\d-]{36}`;
+const mbid = MBID_REGEX_PART;
+
+// List again -stolen- borrowed from kellnerd. Genre doesn't have an edit history.
+export const MB_EDITABLE_ENTITIES = [
+    'area',
+    'artist',
+    'collection',
+    'event',
+    'instrument',
+    'label',
+    'place',
+    'recording',
+    'release',
+    'release-group',
+    'series',
+    'work',
+    'url',
+];
+
 /** Any pages on which edits can occur */
-export const MB_EDIT_PAGE_PATHS = [
-    'edit/*',
-    // <entity>/<entity_id>/edits, user/<username>/edits/open, search/edits?condition.0=..., ...
-    // TODO: This also matches /search/edits, on which no edits are shown. Should somehow be excluded
-    '*/edits*',
-    'user/*/votes',
-    '*/open_edits',
+export const MB_EDIT_DISPLAY_PAGE_PATTERNS = [
+    mb`edit/\d+`,
```

Comment:
```suggestion
    mb`edit/\d+`,
    mb`edit/(open|subscribed(_editors)?)`,
```
More pages with edit listings which I am regularly using.